pkg/ccl/streamingccl/streamclient/client.go

```diff
@@ -147,6 +137,39 @@ func NewStreamClient(streamAddress streamingccl.StreamAddress) (Client, error) {
 	return streamClient, nil
 }
 
+// NewClientFromTopology iterates through each stream address in the topology
+// and returns the first client it's able to successfully connect to.
+func NewClientFromTopology(ctx context.Context, topology *jobspb.StreamTopology) (Client, error) {
+	streamAddresses := make(map[string]struct{})
+	for _, partition := range topology.PartitionInfo {
+		streamAddresses[partition.Address] = struct{}{}
+	}
+
+	var combinedError error = nil
+	for address := range streamAddresses {
```

Comment:
You might want to slow things down (by adding retry struct here) -- often times, if you fail to connect to 1 you'll fail if you re-try immediately.